### PR TITLE
Convert a sorted array to a balanced BST

### PR DESCRIPTION
**Pull request by vishalv971**
https://leetcode.com/problems/convert-sorted-array-to-binary-search-tree/

@SaumyaSachdev / @pranjalkaler 

I am back at it :smile:

### CODE REVIEW

**Comment by vishalv971** [comment]
You see two commits because i tried a local merge.

**Comment by vishalv971** [comment]
> You see two commits because i tried a local merge.

Sorry about that :rofl:

File: LeetCode/CPP/ConvertSortedArrayToBST.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+struct TreeNode {
+    int val;
+    TreeNode *left;
+    TreeNode *right;
+    TreeNode() : val(0), left(nullptr), right(nullptr) {}
+    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+};
```
Comment (pranjalkaler) [inline LeetCode/CPP/ConvertSortedArrayToBST.cpp:16]: You are using dumb pointers 😂 forgot to add a destructor or at least a delete operation.

File: LeetCode/CPP/ConvertSortedArrayToBST.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class Solution {
+    TreeNode* util(std::vector<int> nums, int start, int end ){
+        if(start<=end){
+            int mid = start + (end-start)/2;
+            TreeNode* x = new TreeNode(nums[mid]);
+            x->left = util(nums, start,mid-1);
```
Comment (pranjalkaler) [inline LeetCode/CPP/ConvertSortedArrayToBST.cpp:24]: Are we getting a BST or a regular binary tree here?

File: LeetCode/CPP/ConvertSortedArrayToBST.cpp (hunk 1 above)
Comment (vishalv971) [reply LeetCode/CPP/ConvertSortedArrayToBST.cpp:20]: done done , sorry. Gotta make a common folder with common data structures

File: LeetCode/CPP/ConvertSortedArrayToBST.cpp (hunk 2 above)
Comment (vishalv971) [reply LeetCode/CPP/ConvertSortedArrayToBST.cpp:28]: BST

Comment (pranjalkaler) [reply LeetCode/CPP/ConvertSortedArrayToBST.cpp:28]: in that case, I guess the array is not sorted.

Comment (pranjalkaler) [reply LeetCode/CPP/ConvertSortedArrayToBST.cpp:28]: @vishalv971 approving this one. Please correct this in some later PR 😄

**Review by pranjalkaler — APPROVED** [review]

File: LeetCode/CPP/ConvertSortedArrayToBST.cpp (hunk 2 above)
Comment (vishalv971) [reply LeetCode/CPP/ConvertSortedArrayToBST.cpp:28]: Sorry both of us understood wrong, the question is converting a sorted array to a  BST.

**Pull request merged**